jest-common/src/main/java/io/searchbox/indices/script/CreateIndexedScript.java

```diff
@@ -0,0 +1,107 @@
+package io.searchbox.indices.script;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.searchbox.action.AbstractAction;
+import io.searchbox.action.GenericResultAbstractAction;
+
+import java.io.*;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+
+import static io.searchbox.indices.script.ScriptLanguage.GROOVY;
+import static java.nio.charset.Charset.defaultCharset;
+
+public class CreateIndexedScript extends GenericResultAbstractAction {
+
+  private final String scriptName;
+  private final ScriptLanguage scriptLanguage;
+
+  public CreateIndexedScript(Builder builder) {
+    super(builder);
+    this.payload = builder.payload;
+    this.scriptName = builder.scriptName;
+    this.scriptLanguage = builder.scriptLanguage;
+    setURI(buildURI());
+  }
+
+  protected String buildURI() {
+    try {
+      return super.buildURI() + "/_scripts/" + scriptLanguage.pathParameterName + "/" + URLEncoder.encode(scriptName, CHARSET);
+    } catch (UnsupportedEncodingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public String getRestMethodName() {
+    return "POST";
+  }
+
+  public String getScriptName() {
+    return scriptName;
+  }
+
+  public ScriptLanguage getScriptLanguage() {
+    return scriptLanguage;
+  }
+
+  public static class Builder extends AbstractAction.Builder<CreateIndexedScript, Builder> {
+
+    private String scriptName;
+    private JsonElement payload;
+    private ScriptLanguage scriptLanguage = GROOVY;
+
+    public Builder(String scriptName) {
+      this.scriptName = scriptName;
+    }
+
+    @Override
+    public CreateIndexedScript build() {
+      return new CreateIndexedScript(this);
+    }
+
+    public Builder setSource(String source) {
+      createPayload(source);
+      return this;
+    }
+
+    public Builder loadSource(File srcFile) throws IOException {
```

Comment:
Having `public Builder source(String scriptSource)` would be more than enough (as it has been in other actions) as we are not concerned with taking responsibility of user's streams or files at this point.